index.js

```diff
@@ -0,0 +1,170 @@
+var fs = require('fs-extra');
+var path = require('path');
+var spawn = require('child_process').spawn;
+var hyperquest = require('hyperquest');
+var zlib = require('zlib');
+var unzip = require('unzip');
+var csvToVrt = require('csv-to-vrt');
+var UploadStream = require('./lib/UploadStream');
+var checkHash = require('./lib/checkHash');
+
+var zipReg = /.zip$/i;
+var csvReg = /(?:txt|csv)$/i;
+var restrictedReg = /\.\.|\//;
+
+
+function retrieve(program, callback){
+
+  var scratchSpace = 'retriever-scratch' + Math.random()*1e17;
+  fs.mkdirSync(scratchSpace);
+
+
+  function wrappedCb(err, count){
+    var args = arguments;
+    var self = this;
+
+    fs.remove(scratchSpace, function(e){
```

Comment:
Are you sure you want to remove the scratch space whenever an error occurs?  Any chance there are half-processed files in there you'd want to inspect before they get ditched?